flytekit/extend/image_spec/base_image.py

```diff
@@ -0,0 +1,112 @@
+import base64
+import hashlib
+import json
+import os
+import pathlib
+import subprocess
+from dataclasses import dataclass
+from typing import List, Optional
+
+from dataclasses_json import dataclass_json
+
+IMAGE_LOCK = f"{os.path.expanduser('~')}{os.path.sep}.flyte{os.path.sep}image.lock"
+
+
+@dataclass_json
+@dataclass
+class ImageSpec:
+    """
+    Args:
+        packages: list of packages that will be installed in the image.
+        os: operating system. by default is ubuntu 20.04.
+        registry: docker registry. if it's specified, flytekit will push the image.
+        python_version: python version in the image.
+    """
+
+    packages: List[str]
+    base_image: str = "pingsutw/envd_base:v8"
+    registry: Optional[str] = None
+    python_version: Optional[str] = "3.9"
```

Comment:
let's make this default to the currently running version of python.

Reply:
done